adapters/repos/db/lsmkv/store.go

```diff
@@ -115,17 +117,38 @@ func (s *Store) bucketDir(bucketName string) string {
 func (s *Store) CreateOrLoadBucket(ctx context.Context, bucketName string,
 	opts ...BucketOption,
 ) error {
+	var bucketLock *sync.Mutex
+
+	s.bucketAccessLock.Lock()
```

Comment:
we are actually locking here two times, maybe it would be good to create a method:
```
func (s *Store) lockGuard(mutate func()) {
	p.bucketAccessLock.Lock()
	defer p.bucketAccessLock.Unlock()
	mutate()
}
```
a this operation below and also the one in line 145 we could run with p.lockGuard(...)
wdyt?

Reply:
in this PR there is already a refactoring in progress for this part of the code `https://github.com/weaviate/weaviate/pull/4422`

Given the current code was already validated, we could do the changes as part of that PR...

Reply:
ok, we can do that